### PR TITLE
Adding basic Bigtable client.

### PR DESCRIPTION
**Pull request by dhermes**
FYI @tseaver I am essentially done with the package in https://github.com/dhermes/gcloud-python-bigtable so will now be porting over the code in (hopefully) easy to review chunks.

I'd really like to get this all in by the end of September (end of the quarter).

/cc @jgeewax 

----

**NOTE**: I am holding off on bringing in documentation until there is working code in here.

### CODE REVIEW

File: gcloud/bigtable/client.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    :param admin: (Optional) Boolean indicating if the client will be used to
+                  interact with the Cluster Admin or Table Admin APIs. This
+                  requires the :const:`ADMIN_SCOPE`. Defaults to :data:`False`.
+
+    :raises: :class:`ValueError <exceptions.ValueError>` if both ``read_only``
+             and ``admin`` are :data:`True`
+    """
+
+    # Prevent creation of Client.connection.
+    _connection_class = None
```
Comment (tseaver) [inline gcloud/bigtable/client.py]: Hmm, how does that work?  `Client.__init__()` wants to call `self._connection_class(credentials, http)`.

File: gcloud/bigtable/client.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    def __init__(self, project=None, credentials=None,
+                 read_only=False, admin=False):
+        if read_only and admin:
+            raise ValueError('A read-only client cannot also perform'
+                             'administrative actions.')
+
+        super(Client, self).__init__(project=project, credentials=credentials)
+        del self.http
```
Comment (tseaver) [inline gcloud/bigtable/client.py]: If we are not going to have the `http` attribute expected of `JSONClient` instances, maybe we should not be inheriting from it.

File: gcloud/bigtable/client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigtable/client.py]: OIC, I missed that the base-class changes were below.

Comment (tseaver) [inline gcloud/bigtable/client.py]: I think that makes the case stronger that we shouldn't be using `gcloud.client.Client` or `gcloud.client.JSONClient` as bases:  they are essentially at odds with what we need.

We could factor out the project detection bits from `gcloud.client.JSONClient` for reuse, (and maybe the alternate factories), but we don't really have anything else in common.

Comment (dhermes) [inline gcloud/bigtable/client.py]: I agree. I'll give it a spin and ping you once re-factored.

**Comment by dhermes** [comment]
@tseaver PTAL. I plan on squashing into a single commit before merging, but just added a new commit for now so you could see the diff.

Two concerns:

- By adding `:inherited-members:` to the autodoc for the base client, we end up getting the factories in both the doc for `Client` and `JSONClient`. But this was the only was to liberate them from the non-public mixin
- There is some repeated behavior since I do the `credentials = get_credentials()` check in the Bigtable `Client` constructor. If you've got ideas on how to avoid doing this twice, I'd love to DR(myself)

**Comment by dhermes** [comment]
Also, I'll be teaching until 2pm Pacific tomorrow, then commuting home, but then free for the rest of the week. I'd really like to increase the velocity of these reviews so we can get the Bigtable stuff in by EOQ (it may end up being impossible, but I'd like to at least give it a chance). I've been the bottleneck on the velocity so far.

File: gcloud/client.py [hunk 3]
```diff
@@ -115,16 +99,14 @@ def from_service_account_p12(cls, client_email, private_key_path,
...
-    value.
+    Assumes that the associated ``_connection_class`` only accepts
+    ``http`` and ``credentials`` in its constructor.
 
-    :type project: string
-    :param project: the project which the client acts on behalf of. If not
-                    passed falls back to the default inferred from the
-                    environment.
+    If ``_connection_class`` is :data:`None`, stores the ``credentials``
+    on the client instead of the newly created ``connection``.
```
Comment (tseaver) [inline gcloud/client.py]: No longer true for this class.

**Comment by tseaver** [comment]
LGTM, modulo the note about the out-of-date docstring for `gcloud.client.Client`.

**Comment by dhermes** [comment]
@tseaver I ripped out the comment and squashed the commits into a single one.

Merging now. LMK if there are issues and we can roll back.

**Pull request merged**